src/swapd/process_msgs.rs

```diff
@@ -0,0 +1,1043 @@
+use std::{
+    collections::HashMap,
+    time::{Duration, SystemTime},
+};
+
+use crate::{
+    databased::checkpoint_send,
+    event::Event,
+    rpc::{
+        request::{
+            self, BitcoinFundingInfo, CheckpointState, FundingInfo, InitSwap, Msg, Outcome, Params,
+            Reveal, Tx,
+        },
+        Failure, FailureCode, Request, ServiceBus,
+    },
+    swapd::{
+        runtime::Runtime,
+        swap_state::{AliceState, BobState},
+        CheckpointSwapd, SwapCheckpointType,
+    },
+    syncerd::{bitcoin_syncer::p2wpkh_signed_tx_fee, SweepMoneroAddress, XmrAddressAddendum},
+    CtlServer, Endpoints, Error, LogStyle, ServiceId,
+};
+use farcaster_core::{
+    blockchain::Blockchain,
+    role::{SwapRole, TradeRole},
+    swap::btcxmr::message::CoreArbitratingSetup,
+    transaction::TxLabel,
+};
+use microservices::esb::{self, Handler};
+
+use super::{
+    get_swap_id,
+    runtime::{
+        aggregate_xmr_spend_view, remote_params_candidate, PendingRequest, PendingRequests,
+        PendingRequestsT,
+    },
+    swap_state::Awaiting,
+    syncer_client::log_tx_received,
+    State,
+};
+
+impl Runtime {
```

Comment:
The `process_msgs.rs` file only excutes on the runtime. There are now a total of 5 `impl Runtime` within swapd and I don't think this separation makes sense. Either you separate them with abstractions and actually process them with `next()` on the state machine ( StateMachine -> ( process() ) -> StateMachine ) or add them back in the service runtime.